bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -248,23 +323,54 @@ protected NeoHubReadDcbResponse fromNeoHubReadDcbResponse() {
      * handlers
      */
     private synchronized void lazyPollingSchedulerExecute() {
-        NeoHubInfoResponse infoResponse = fromNeoHubReadInfoResponse();
-
-        if (infoResponse != null) {
-            // determine temperatureUnit
-            NeoHubReadDcbResponse dcbResponse = fromNeoHubReadDcbResponse();
-            Unit<?> temperatureUnit = (dcbResponse != null) ? dcbResponse.getTemperatureUnit() : SIUnits.CELSIUS;
+        // check which API is supported
+        if (!isApiOnline) {
+            selectApi();
+        }
 
-            // dispatch infoResponse to each of the hub's owned devices ..
+        NeoHubAbstractDeviceData deviceData = fromNeoHubGetDeviceData();
+        if (deviceData != null) {
+            // dispatch deviceData to each of the hub's owned devices ..
             List<Thing> children = getThing().getThings();
             for (Thing child : children) {
                 ThingHandler device = child.getHandler();
                 if (device instanceof NeoBaseHandler) {
-                    ((NeoBaseHandler) device).toBaseSendPollResponse(infoResponse, temperatureUnit);
+                    ((NeoBaseHandler) device).toBaseSendPollResponse(deviceData);
                 }
             }
-        }
 
+            // evaluate and update the state of our RF mesh QoS channel
+            List<? extends AbstractRecord> devices = deviceData.getDevices();
+            State state;
+
+            if (devices == null || devices.isEmpty()) {
+                state = UnDefType.UNDEF;
+            } else {
+                int totalDeviceCount = devices.size();
+                int onlineDeviceCount = 0;
+
+                for (AbstractRecord device : devices) {
+                    String deviceName = device.getDeviceName();
+                    Boolean online = !device.offline();
+
+                    if (connectionStates.containsKey(deviceName)) {
+                        @Nullable
+                        Boolean onlineBefore = connectionStates.get(deviceName);
+                        if (!online.equals(onlineBefore)) {
+                            logger.info("device \"{}\" has {} the RF mesh network", deviceName,
+                                    online.booleanValue() ? "joined" : "left");
+                        }
+                    }
+                    connectionStates.put(deviceName, online);
+
+                    if (online.booleanValue()) {
+                        onlineDeviceCount++;
+                    }
+                }
+                state = new DecimalType((100.0 * onlineDeviceCount) / totalDeviceCount);
```

Comment:
Shouldn't this be a quantity type with Percent units?

Reply:
Yes. Done. But it reminds me that I should also have used QuantityType Percent on the [siemensrds] humidity point so I will do that as well and make new a "micro" PR for it.

Reply:
Yes. Done. But it reminds me that I should also have used QuantityType Percent on the [siemensrds] humidity point so I will do that as well and make new a "micro" PR for it.